functions/Errors/ErrorInScriptResult.fmfn

```diff
@@ -0,0 +1,54 @@
+/**
+ * =====================================
+ * ErrorInScriptResult ( )
+ *
+ * RETURNS:
+ *		Boolean result. Raises $error and $result based on ScriptResult and LastError.
+ *
+ * PARAMETERS:
+ *		None.
+ *
+ *
+ * DEPENDENCIES:
+ *		Error(), ErrorFmp(), ErrorFound ()
+ *
+ * HISTORY:
+ *		CREATED on 2023-DEC-23 by Robert Naud robert@accoladeplusaccolade.com
+  *
+ * REFERENCES:
+ *		https://github.com/filemakerstandards/fmpstandards/tree/master/Functions/Errors
+ * =====================================
+ */
+
+Let ( [
+~error.code = ErrorFmp ( "" ) ;
+~result = Get ( ScriptResult ) ;
+~result.json.isValid = Left ( JSONGetElement ( ~result ; "" ) ; 3 ) ≠ "? *" ;
+~error = Case (
+	~error.code ;
+		$error ;
+	not ~result.json.isValid ;
+		Error ( "app" ; -2 ; "Script result is not valid JSON." ; Get ( ScriptResult ) ) ;
```

Comment:
I think this should use the `ErrorApp` CF instead of `Error`.

When I created the concept of "app" errors, I always intended for common errors like this to be defined in the `ErrorApp` CF itself. I'm not sure you _should_ standardize on the codes I use, but to give an example at least, here's the codes I define in my `ErrorApp` CF:

```
theErrorCode = 0 ;     "No error" ;
theErrorCode = 1 ;     "User canceled action" ;
theErrorCode = 2 ;     "Invalid Parameter" ; /* Missing or malformed, so if a date is expected but script received a turtle, parameter is invalid. */
theErrorCode = 3 ;     "Failed Validation" ; /* Any general-purpose validation that doesn't fit into another more-specific error code. */
theErrorCode = 6 ;     "Invalid window mode" ;
theErrorCode = 10 ;     "Incompatible FileMaker version" ; /* Sometimes we offer a feature or two that doesn't run on the lowest supported client version. In that case, this error could be used to only allow supported clients to run the code. An example of this is the Export All script. */
theErrorCode = 11 ;     "Incompatible FileMaker application" ; /* Might use if FMGo is required for a GPS location based function, but the feature was run from FMPro. */
theErrorCode = 13 ;     "Script Error" ; /* I often wonder if I should use this code, or Failed Validation code 3. I think there are subtle differences, though. Let's say a script set a variable to the value of "a", "b" or "c", then later tested which value it was. I would throw this code in an Else since the same script set the code, so there is just some internal error that isn't because of a parameter. It might be able to be called validation error, but the script itself is what was invalid, so I use this code.*/
theErrorCode = 16 ;     "Plug-In error" ; /* Use when a plugin doesn't have a specific error handling function (which set's the type to the plugin name) */
theErrorCode = 17 ;     "ExecuteSQL error" ;
theErrorCode = 19 ;     "Invalid script result" ; /* Set by #AssignScriptResultAndSetError script. Yes, it's basically like code 2, but for script results. You could also use it when translating a module's result to our standard, and that process fails. */

// else:
"Unknown error code"
```

You could probably replace 6, 10, and 11 with 3 (Failed Validation) but in my `ErrorApp` CF, I add relevant data to the error object for these codes: (note that this is from my old let-notation based CF)

```
Error ( "app" ; theErrorCode ; errorDescription ; theErrorInfo )
& Case (
	theErrorCode = 6 ;     # ( "WindowMode" ; Get ( WindowMode ) ) ;
	theErrorCode = 10 ;    # ( "ApplicationVersion" ; Get ( ApplicationVersion ) ) ;
	theErrorCode = 11 ;    # ( "ApplicationVersion" ; Get ( ApplicationVersion ) )
)
```

Reply:
The CF was likely written before `ErrorApp (  )` existed. Calling the `Error (  )` CF directly also diminishes the CF interdependencies for devs who would implement without the `ErrorApp(  )` CF.

That said, if we see this as a minimal set of custom functions to be used together and we do have `ErrorApp` as part of the toolkit, then we should leverage it indeed.

Reply:
I fully agree we should limit the number of interconnections between the functions to limit dependencies. Given we're shooting for Best Practices, based on the knowledge of experienced devs, it's still important that we can support a minimal implementation with the ability to expand as desired or adapt from what's provided.

Given the above thought process, I wonder if we should revise the folder structure to the following to distinguish from the fact that ErrorApp, ErrorIf, ErrorLog, ErrorFound and ErrorScriptResult are methods which may or may not be preferred by some devs. 

- Errors
  - Core
    - Error.fmfn
    - ErrorDetails.fmfn
    - ErrorEnvironment.fmfn
    - ErrorFmp.fmfn
  - Extras
    - ErrorApp.fmfn
    - ErrorFound.fmfn
    - ErrorIf.fmfn
    - ErrorLog.fmfn
    - ErrorScriptResult.fmfn

Reply:
I would consider ErrorInScriptResult to be part of the core functions, but it's simply a question about where we trace the line, just like some people may see ErrorEnvironment, ErrorDetails or ErrorFMP as extras. I like the concept of labeling some as core and some as extras.

I don't want to sound picky. I'm ok with any classification we come up with.

Or we can kick this down, do a first release and do a revision once we land on where to trace the line. All of those options are fine by me.

The core/extra concept was absent in the original functions. Would we say it still applies to the original set or was the original set all considered core?

Reply:
> Or we can kick this down, do a first release and do a revision once we land on where to trace the line. All of those options are fine by me.

This makes sense to me. I have some pretty strong opinions on error trapping and I wouldn't mind discussing the merits of those opinions and if they should sway any code in this repo. I don't think this pull request is the best place to have that conversation, though. Maybe we can start a discussion on filemakerstandards.org?

I haven't reviewed a pull request like this in GitHub before so I've likely broken all known conventions/best practices in doing so, and I'm a little lost as to the current state of things.

Reply:
I don't do enough things on GitHub to be aware of conventions. When comments led to modifications I could rally to, I made the modifications. On some other things, I see the conversations as more open-ended. There are some modifications that I would rather not make in my own fork / pull request, but can be totally fine with me if it was to be altered once it got in the main branch (or a candidate branch).

To have the discussion on filemakerstandards.org, I do feel like some of the recent changes need to out of the pull request state. I don't think we want discussions there to point to a specific pull request.

The code by itself works. There could be kinks to iron out, but it's not like pushing something that does not work and is a work in progress.

It should be easier to involve others in the discussion over at fmpstandards.org